index.js

```diff
@@ -169,10 +169,7 @@ function fastifyMultipart (fastify, options, done) {
     })
   }
 
-  let throwFileSizeLimit = true
-  if (typeof options.throwFileSizeLimit === 'boolean') {
-    throwFileSizeLimit = options.throwFileSizeLimit
-  }
+  const defaultThrowFileSizeLimit = options.throwFileSizeLimit !== false
```

Comment:
The check of `typeof options.throwFileSizeLimit === 'boolean'` should be keep.

Reply:
Because it could result in a deoptimization?

Reply:
```suggestion
  const defaultThrowFileSizeLimit = typeof options.throwFileSizeLimit === 'boolean'
    ? options.throwFileSizeLimit
    : true
```